src/engraving/rw/read114/read114.cpp

```diff
@@ -2247,14 +2245,29 @@ static void readBox(XmlReader& e, ReadContext& ctx, Box* b)
             HBox* hb = Factory::createHBox(bSystem);
             readBox(e, ctx, hb);
             b->add(hb);
+            keepMargins = true;           // in old file, box nesting used outer box margins
         } else if (tag == "VBox") {
             VBox* vb = Factory::createVBox(bSystem);
             readBox(e, ctx, vb);
             b->add(vb);
+            keepMargins = true;           // in old file, box nesting used outer box margins
         } else if (!readBoxProperties(e, ctx, b)) {
             e.unknown();
         }
     }
+
+    // with .msc versions prior to 1.17, box margins were only used when nesting another box inside this box:
+    // for backward compatibility set them to 0.0 in all other cases, the Mu1 defaults of 5.0 just look horrible in Mu3 and Mu4
+
+    if (ctx.mscVersion() <= 114 && (b->isHBox() || b->isVBox()) && !keepMargins) {
+        b->setLeftMargin(0.0);
+        b->setRightMargin(0.0);
+        b->setTopMargin(0.0); // 2.0 would look closest to Mu1 and Mu2, but 0.0 is the default since Mu2
+        b->setBottomMargin(0.0); // 1.0 would look closest to Mu1 and Mu2, but 0.0 is the default since Mu2
+        if (b->bottomGap() != Spatium().toMM(7.0)) { // Default was 1.0, but above 6.0 are added for Mu1 scores
```

Comment:
> // Default was 1.0, but above 6.0 are added for Mu1 scores

I don't really get what you mean here...

Also, `Spatium().toMM(7.0)` looks really weird to me. It means: create an instance of the `Spatium` class (which means "zero spatium") and then convert that to millimetres, assuming that one spatium is 7.0 millimetres. That is absolutely not what you want here :)
Instead, how about `b->propertyDefault(Pid::BOTTOM_GAP)`?

Reply:
See line 3097:
```cpp
    masterScore->style().set(Sid::frameSystemDistance, masterScore->style().styleS(Sid::frameSystemDistance) + Spatium(6.0));
```

Reply:
I don't get what you mean about using `b->propertyDefault(Pid::BOTTOM_GAP)`, how?
`if (b->bottomGap() != b->propertyDefault(Pid::BOTTOM_GAP))` doesn't work

Reply:
Hmm, maybe
`if (b->bottomGap() != b->propertyDefault(Pid::BOTTOM_GAP).value<Millimetre>())`

Reply:
Yes indeed, sorry.
I assume what you want to do is checking whether the value after reading matches the current style default, and if not, mark the prop as unstyled. Right?

Reply:
Rather whether it was the old Mu1 default. But that got this 6.0 added automagicallly, as far as I can tell

Reply:
But actually, that should be redundant, because `readBoxProperties` (which calls `read400::TRead::readProperty`) already takes care of setting property flags correctly for values that don't match their default. So this entire `if` block would do nothing as far as I understand.

Reply:
And if it's _not_ redundant for some reason, then I wouldn't see why this should be inside the `!keepMargins` block.

Reply:
But that ready the actuall setting so possible the old default of 1.0, and then makses it styled, That's what I want to fix

Reply:
Okay, I think I understand it slightly better now:
- In MS1, when `bottomGap` was at its default, it is not written to the file, of course
- Thus, during reading in MS4, it is not encountered
- Thus, it stays at the default that MS4 initialised it to, and stays STYLED
- "The default that MS4 initialised it to" means: the value from `Sid::frameSystemDistance` _at the time that the Box is instantiated_.
- At the time that the Box is instantiated, that style value is the value that was read while reading the `<Style>` tag, but the values in that tag had a different meaning for MS1. 
- Therefore, later, the `+ Spatium(6.0)` is done. But that is too late, because the Box is instantiated before that; so the Box contains the wrong default.

That's the problem, right?

In that case, my `b->propertyDefault(Pid::BOTTOM_GAP)` ain't gonna work either indeed, because at the point that that's called, the style value returned by it has not yet been updated.

I think the real solution would be to ensure that the style values are updated directly after reading them, before any EngravingItem is initialised. I.e. move those lines under `// adjust some styles` into the `else if (tag == "Style") {` block.

Reply:
Makes sense, let's try that

Reply:
Hmm, actually now I'm completly confised and don't even know anymore what I really wanted to fix.
So let's just drop that one for now and restring to resetting the frame borders (by moving the code from read2006.cpp to read114.cpp and removing apparently redundant stuff)

Reply:
I wonder what that does to the mtest and vtests...

Reply:
OK, one mtest file was affected, and for good reason.

Actually the issue I wanted to fix was an issue with 2.x importing 1.x files, doesn't happen with 3.x nor 4.x
It does seem to happen though when importing a 2.x score into 3.x or 4.x, if that score originally stemmed from 1.x (and still had that property marked as unstyled), 2.x set the bottom gap to 7.0 (and unstyled) 3.x and 4.x set it to 8.0 (!?!) and still unstyled. So if anything this issue would need to get fixed in read206.cpp

Reply:
If you got some spare time and are curious, check the attached
[Frametests.zip](https://github.com/user-attachments/files/16325165/Frametests.zip)